### ISSUE TITLE
Translation for Server-Side Messages

### ISSUE THREAD

**Issue by oruburos** [processing/p5.js-web-editor#1587]
- Existing feature enhancement

Messages from server-side operations are not translated
For instance:

>  https://github.com/processing/p5.js-web-editor/blob/develop/server/controllers/user.controller.js#L60-L61

> https://github.com/processing/p5.js-web-editor/blob/develop/server/controllers/user.controller.js#L147

> https://github.com/processing/p5.js-web-editor/blob/develop/server/controllers/user.controller.js#L176
#### Feature enhancement details:
In order to not break the API, maybe introducing an extra field in the response to identify the string to display,  and the field that triggers that response, and get the translation in the client, according to the language of the client.

 For example , an extract from these lines 
https://github.com/processing/p5.js-web-editor/blob/develop/server/controllers/user.controller.js#L102-L108

```

 if (user) {
      return res.json({
        exists: true,
        message: `This ${checkType} is already taken.`,
        type: checkType
      });
    }
```

would be

```

 if (user) {
      return res.json({
        exists: true,
        message: `This ${checkType} is already taken.`,
        messageKey: ServerTranslation.UserTaken,
        type: checkType
      });
    }

```

**Comment by catarak** [comment]
This makes sense to me! We could also just remove the `message` key and have all of the text be handled client-side.

**Comment by oruburos** [comment]
@catarak yes, I agree, but I only had the perspective from the web editor, I didn't know if there were other clients expecting the message field from the server.

**Comment by catarak** [comment]
> I didn't know if there were other clients expecting the message field from the server.

The API is only being used by the web editor right now!

**Comment by andrewn** [comment]
I was wondering if for the public API, we would want the English language message in addition to the messageKey?

**Comment by catarak** [comment]
> I was wondering if for the public API, we would want the English language message in addition to the messageKey?

That makes sense to me! Perhaps it would make sense to open a separate issue about how to handle translations with the public API?
